pagefaultgames/pokerogue
# [QoL] Modifiers type inference

## What are the changes?
Add type inference to all modifier classes

## Why am I doing these changes?
`as Type` assertions are dangerous and this make the user experience better by adding intellisense

## What did change?
1. Make applyModifier method accept generic modifier classes and get typeof args from inference
2. Refactor all places modifiers are used to follow the new type inference param scheme
3. Create a new utility type that maps to `new(...args: any[]) => SomeClass` 
4. Refactor all places that used `new(...args: any[]) => SomeClass` to now use `Utils.ClassType<T>`

## How to test the changes?
Try it out with intellisense.
Run the game in localhost to see the changes aren't breaking

## Checklist
  - The `Utils.ClassType<T>` alias can be its own separate PR 

~~- [ ] Are the changes visual?~~
~~- [ ] Have I provided screenshots/videos of the changes?~~
